src/main/kotlin/se/uulm/snowballr/backend/model/SnowballRException.kt

```diff
@@ -143,6 +144,30 @@ sealed class SnowballRException(
         ) : InvalidIdException(entityType, id, "UUID")
     }
 
+    /**
+     * Represents a specific type of exception that occurs a value is not in the correct range.
+     *
+     * @param value The value that is not in range.
+     * @param from The left border of the correct range interval.
+     * @param to The right border of the correct range interval.
+     */
+    sealed class OutOfRangeException(
+        value: Float,
+        from: Float,
+        to: Float,
+    ) : SnowballRException("The value $value is not in the range of from $from to $to.") {
+        /**
+         * Represents an [OutOfRangeException] that occurs when a [Stage] value is not in the correct range.
+         *
+         * @param stage The value of the stage.
+         * @param maxStage The maximum value of the stages allowed.
+         */
+        class Stage(
+            stage: Long,
+            maxStage: Long,
+        ) : OutOfRangeException(stage.toFloat(), 0F, maxStage.toFloat())
```

Comment:
Here you can use `Number` as a general type for numerical values. This way, you don't have to convert the long values to floats.

```suggestion
        value: Number,
        from: Number,
        to: Number,
    ) : SnowballRException("The value $value is not in the range of from $from to $to.") {
        /**
         * Represents an [OutOfRangeException] that occurs when a [Stage] value is not in the correct range.
         *
         * @param stage The value of the stage.
         * @param maxStage The maximum value of the stages allowed.
         */
        class Stage(
            stage: Long,
            maxStage: Long,
        ) : OutOfRangeException(stage, 0, maxStage)
```

Reply:
Good call. Didn't recognize that kotlin nows `Number`.